src/test/java/seedu/address/testutil/PersonBuilder.java

```diff
@@ -98,8 +98,8 @@ public PersonBuilder withEmail(String email) {
     /**
      * Sets the {@code Department} of the {@code Person} that we are building.
      */
-    public PersonBuilder withDepartment(String department) {
-        this.department = Optional.of(new Department(department));
+    public PersonBuilder withDepartment(Optional<Department> department) {
+        this.department = department;
```

Comment:
If you're gonna be adding a department, you don't need an optional, I think you're inverting the responsibility here. Instead it should be something like.
```suggestion
    public PersonBuilder withDepartment(Department department) {
        requireNonNull(department);
        this.department = Optional.of(department);
```

Reply:
And when you don't want to add a department, just don't call the function.

Reply:
Oh and of course, make sure that the department is `Optional.empty()` by default for my suggestion to work.